gen/MotController_data.yml

```diff
@@ -120,23 +122,40 @@ c_MotController_Create1:
       # -> 0/1 refers to pidIdx
       'pid0_iaccum': 0.0,
       'pid0_error': 0,
+      'pid0_prevErr': 0,
       'pid0_errorDerivative': 0.0,
       'pid0_feedback': FeedbackDevice.None_,
+      'pid0_notFirst': False,
+      'pid0_outBeforRmp': 0,
+      'pid0_target': 0,
       
       'pid1_iaccum': 0.0,
       'pid1_error': 0,
+      'pid1_prevErr': 0,
       'pid1_errorDerivative': 0.0,
       'pid1_feedback': FeedbackDevice.None_,
+      'pid1_notFirst': False,
+      'pid1_outBeforRmp': 0,
+      'pid1_target': 0,
       
       # slotIdx
       'profile_slot_select': 0,
       # pidIdx
       'pid_slot_select': 0,
       
+      # motion magic parameters
+      'motionmagic_cruise_velocity': 0,
+      'motionmagic_acceleration': 0,
+      'motionmagic_target': 0,
+      'motionmagic_velocity': 0,
       
     })
   
     hal_data['CAN'][deviceId] = self.hal_data
+
+    if self.Notifier:
```

Comment:
Shouldn't this only run when magic/pid mode is enabled?

Reply:
_calculate_1ms won't do anything unless its in magic/pid mode. I suppose you could enable/disable in setDemand.

Reply:
That would be ideal; no sense creating more threads if they aren't needed.

Reply:
not at a place where I can check easily, is there a free function somewhere in the talon class? that notifier is probably going to have to be stopped at some point

Reply:
The notifier has a free function and will stop itself.